coalib/results/Diff.py

```diff
@@ -70,6 +71,23 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unidiff(cls, diff_array):
+        """
+        Creates a Diff object from given unified diff.
+        :param diff_array: list of lines of unified diff.
+        """
+        source_array = []
+        target_array = []
+        patch_set = unidiff.PatchSet(diff_array)
+        for patched_file in patch_set:
+            for hunk in patched_file:
+                for line in hunk.source_lines():
+                    source_array.append(str(line)[1:])
```

Comment:
> one never gets a complete file in a unified diff

Not necessarily, I guess it depends on the unified diff itself. But why do we need complete files to generate the `Diff` object?